### ISSUE TITLE
Release 4.4.0 - Beta 1 - E2E UX tests - Amazon Cloudtrail integration

### ISSUE THREAD

**Issue by chemamartinez** [wazuh/wazuh#15895]
The following issue aims to run the specified test for the current release candidate, report the results, and open new issues for any encountered errors.

## Test information
|                         |                                            |
|-------------------------|--------------------------------------------|
| **Test name**           | Amazon Cloudwatch Integration |
| **Category**            | Cloud Security |
| **Deployment option**   |Single indexer and dashboard, multiple servers: Quickstart|
| **Main release issue**  |        https://github.com/wazuh/wazuh/issues/15893                                    |
| **Release candidate #** |                Beta 1                            |
| **Previous issue** |  https://github.com/wazuh/wazuh/issues/15770  |

## Test description
Test different use cases of Amazon events Cloudwatch and see that alerts are generated and make sense.

|                         |     |           |                                    |              |    |
|-------------------------|-----|-----------|------------------------------------|--------------|----|
| Installation            | RAM | Disk      | Cores                              | IP           | OS |
|Indexer,Manager,Dashboard| 4Gb | 30Gb NVMe | 2 processors,2 cores per processor | 192.168.0.37 | Red Hat Enterprise Linux 8.1 clean install |
| Agent | 2Gb | 20Gb | 1 processor, 1 core per processor | 192.168.0.53 | Red Hat Enterprise Linux 8.1 clean install |

## Test report procedure

All test results must have one of the following statuses: 
|                                  |                                            |
|---------------------------------|--------------------------------------------|
| :green_circle:  | All checks passed. |
| :red_circle:  | There is at least one failed result. |
| :yellow_circle:  | There is at least one expected failure or skipped test and no failures. |

Any failing test must be properly addressed with a new issue, detailing the error and the possible cause. 

An extended report of the test results must be attached as a ZIP or TXT file. Please attach any documents, screenshots, or tables to the issue update with the results. This report can be used by the auditors to dig deeper into any possible failures and details.

## Conclusions

All tests have been executed and the results can be found [here]().

|            |                                |                     |                |
|------------|--------------------------------|---------------------|----------------|
| **Status** |     **Test**                   | **Failure type**    | **Notes**      |
|    🟢     | [Wazuh Manager Quickstart](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1387562628) |          N/A        |                |
|    🟡     | [Wazuh Agent Installation](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396087896) | Ports must be opened & domain name had to be changed | Although not critical, no advice is in the documentation about these 2 minor installation problems |
|    🟡     | [Monitoring AWS infrastructure](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396172299) | Permissions problems are reported as credential problems at log which might lead to incorrect diagnostics | Improvement on log text can simplify solving a permission issue |
|           |                      | Documentation marks as deprecated a working authentication method |  |
|    🟡     | [EC2: Run a new instance](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396198689) |          No log appeared at Wazuh's dashboard nor at AWS Cloudtrail either        |       Already reported at [#issue 15807](https://github.com/wazuh/wazuh/issues/15807)         |
|    🟡     | [EC2: Start instances](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396201367) |          N/A        |       Log appeared only after restarting Wazuh server         |
|    🟡    | [EC2: Stop instances](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396215116) |          N/A        |       Log appeared only after restarting Wazuh server         |
|    🟡     | [EC2: Create Security Groups](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396224590) |          N/A        |       Log appeared only after restarting Wazuh server         |
|    🟡    | [EC2: Allocate a new Elastic IP address](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396354161) |          N/A        |       Log appeared only after restarting Wazuh server         |
|    🟡    | [EC2: Associate a new Elastic IP address](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396363890) |          N/A        |       Log appeared only after restarting Wazuh server         |
| | | | |
|    🟡   | [IAM: Create a user account](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396367620) |          N/A        |        Log appeared only after restarting Wazuh server        |
|         | [IAM: Create a user account without permissions](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396373918) |          N/A        |       Log appeared only after restarting Wazuh server         |
|    🟡   | [IAM: User login failed](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396375476) |          N/A        |         Log appeared only after restarting Wazuh server       |
|    🟡    | [IAM: Possible break-in attempt](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396379579) |          No log appeared at Wazuh's dashboard nor at AWS Cloudtrail either       |        Already reported at [#issue 15807](https://github.com/wazuh/wazuh/issues/15807)        |
|    🟡   | [IAM: Login success](https://github.com/wazuh/wazuh/issues/15895#issuecomment-1396380176) |          N/A        |       Log appeared only after restarting Wazuh server         |

All tests have passed and the fails have been reported or justified. Therefore, I conclude that this issue is finished and OK for this release candidate.

## Auditors validation
The definition of done for this one is the validation of the conclusions and the test results from all auditors.

All checks from below must be accepted in order to close this issue.

- [x] @chemamartinez 
- [ ] @wazuh/framework  
- [ ] @davidjiglesias

**Comment by ncvicchi** [comment]
# 🟡 [Wazuh Agent Installation](https://documentation-dev.wazuh.com/v4.4.0-beta1/installation-guide/wazuh-agent/index.html)

NOTE: After deployment finished, the agent was not able to enroll to Manager. 2 causes were responsibles:

- RHEL firewall was blocking connections. It had to be disabled

```console

systemctl stop firewalld.service

```

- Since both virtual machines had identical OS clean installations, both had the same hostname, which prevents enrollment. The hostname at `/etc/hostname` in the agent machine was change from `localhost.localdomain` a `localhost.localagent`.

## Add the Wazuh repository

<details><summary>Import GPG Key</summary>
---

```console

rpm --import https://packages-dev.wazuh.com/key/GPG-KEY-WAZUH

```
</details>

<details><summary>Add repository</summary>

```console

cat > /etc/yum.repos.d/wazuh.repo << EOF
[wazuh]
gpgcheck=1
gpgkey=https://packages-dev.wazuh.com/key/GPG-KEY-WAZUH
enabled=1
name=EL-\$releasever - Wazuh
baseurl=https://packages-dev.wazuh.com/pre-release/yum/
protect=1
EOF

```
</details>

<details><summary>Deploy Wazuh Agent</summary>

```console

WAZUH_MANAGER="192.168.0.37" yum install wazuh-agent

```
</details>

<details><summary>Enable and start the Wazuh agent service</summary>

```console

systemctl daemon-reload
systemctl enable wazuh-agent
systemctl start wazuh-agent

```
</details>

<details><summary>Recommended action - Disable Wazuh updates</summary>

```console

sed -i "s/^enabled=1/enabled=0/" /etc/yum.repos.d/wazuh.repo

```

</details>

**Comment by ncvicchi** [comment]
# 🟡 [Monitoring AWS infrastructure](https://documentation-dev.wazuh.com/v4.4.0-beta1/proof-of-concept-guide/aws-infrastructure-monitoring.html)

#### Cloudtrail was configured at AWS according to the [documentation](https://documentation-dev.wazuh.com/v4.4.0-beta1/proof-of-concept-guide/aws-infrastructure-monitoring.html)

<details><summary>Enable AWS module at dashboard</summary>

![image](https://user-images.githubusercontent.com/69121070/213305056-a5499baf-dfa3-460a-bc5a-4b8aa2ada390.png)

</details>

<details><summary>Create /root/.aws/credentials file with access profiles</summary>

```console

[root@localhost manager]# cat /root/.aws/credentials 
[default]
aws_access_key_id=********************
aws_secret_access_key=****************************************
region=us-east-1

[dev]
aws_access_key_id=********************
aws_secret_access_key=****************************************
region=us-east-1

[prod]
aws_access_key_id=********************
aws_secret_access_key=****************************************
region=us-east-1


```

</details>

<details><summary>Enable AWS module at ossec.conf file</summary>

```console

<wodle name="aws-s3">
  <disabled>no</disabled>
  <remove_from_bucket>no</remove_from_bucket>
  <interval>30m</interval>
  <run_on_start>yes</run_on_start>
  <skip_on_error>no</skip_on_error>

  <bucket type="cloudtrail">
    <name>aws-cloudtrail-logs-567970947422-f4581f65<</name>
    <aws_profile>prod</aws_profile>
  </bucket>
</wodle>

```

Restart wazuh-server:

```console

sudo systemctl restart wazuh-manager

```

</details>

<details><summary>Test the configuration</summary>

Events can be seen at dashboard:

![image](https://user-images.githubusercontent.com/69121070/213306282-d176e82e-5097-44c0-9710-013533eeea4f.png)

</details>

<details><summary>:yellow_circle: Log issue</summary>

A misconfiguration on permissions of the AWS user used for the tests shows at the log like this:

```console

2023/01/18 03:48:52 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  Returned exit code 3
2023/01/18 03:48:52 wazuh-modulesd:aws-s3: WARNING: Bucket:  -  Invalid credentials to access S3 Bucket

```

Although strictly correct, this error log might confuse the user into believing that the credentials are incorrect, while only permissions of the users are to be corrected.

</details>

<details><summary>:yellow_circle: Documentation issue</summary>

Documentation states that setting credentials directly at the ossec.conf file instead of using references to profiles at /root/.aws/credentials is deprecated in version 4.4. Instead, this method was probed to be working.

</details>

**Comment by ncvicchi** [comment]
# :green_circle: [Run a new instance in EC2]
(https://documentation.wazuh.com/current/amazon/services/supported-services/cloudtrail.html#run-a-new-instance-in-ec2)

#### Test performed successfully.

<details><summary>Evidence</summary>

![image](https://user-images.githubusercontent.com/69121070/213312278-61b42353-224d-4cc5-84b0-3a86fbeb1af2.png)

</details>

**Comment by ncvicchi** [comment]
# :green_circle: [Start instances in EC2](https://documentation.wazuh.com/current/amazon/services/supported-services/cloudtrail.html#start-instances-in-ec2)

#### Test performed successfully.

<details><summary>Evidence</summary>

![image](https://user-images.githubusercontent.com/69121070/213312278-61b42353-224d-4cc5-84b0-3a86fbeb1af2.png)

</details>

**Comment by ncvicchi** [comment]
# :green_circle: [Stop instances in EC2](https://documentation.wazuh.com/current/amazon/services/supported-services/cloudtrail.html#stop-instances-in-ec2)

#### Test performed successfully.

<details><summary>Evidence</summary>

![image](https://user-images.githubusercontent.com/69121070/213315114-10c43554-aecc-47a3-abd0-cf00adf5c1eb.png)

</details>

**Comment by ncvicchi** [comment]
# :yellow_circle: [EC2: Create Security Groups](https://documentation.wazuh.com/current/amazon/services/supported-services/cloudtrail.html#create-security-groups-in-ec2)

#### Log appeared only after restarting Wazuh server

<details><summary>Evidence</summary>

![image](https://user-images.githubusercontent.com/69121070/213317098-71941a60-ff75-43b2-9ee4-e9204846c722.png)

</details>

**Comment by ncvicchi** [comment]
# :yellow_circle: [EC2: Allocate a new Elastic IP address](https://documentation.wazuh.com/current/amazon/services/supported-services/cloudtrail.html#allocate-a-new-elastic-ip-address)

#### Log appeared only after restarting Wazuh server

<details><summary>Evidence</summary>

![image](https://user-images.githubusercontent.com/69121070/213339433-7fbcc4d1-e901-418f-bdb9-cbf8a8660149.png)


</details>

**Comment by ncvicchi** [comment]
# :yellow_circle: [EC2: Associate a new Elastic IP address](https://documentation.wazuh.com/current/amazon/services/supported-services/cloudtrail.html#associate-a-new-elastic-ip-address)

#### Log appeared only after restarting Wazuh server

<details><summary>Evidence</summary>

![image](https://user-images.githubusercontent.com/69121070/213341538-0a1255ca-5e3f-44b1-abd1-4da7b6e7c886.png)

</details>

**Comment by ncvicchi** [comment]
# :yellow_circle: [IAM: Create a user account](https://documentation.wazuh.com/current/amazon/services/supported-services/cloudtrail.html#create-a-user-account)

#### Log appeared only after restarting Wazuh server

<details><summary>Evidence</summary>

![image](https://user-images.githubusercontent.com/69121070/213342549-14e6f70d-c258-468c-a1c8-ccb6ff581eb7.png)

</details>

**Comment by ncvicchi** [comment]
# :red_circle: [IAM: Create a user account without permissions](https://documentation.wazuh.com/current/amazon/services/supported-services/cloudtrail.html#create-a-user-account-without-permissions)

#### No log appeared at Wazuh's dashboard nor at AWS Cloudtrail either

**Comment by ncvicchi** [comment]
# :yellow_circle: [IAM: User login failed](https://documentation.wazuh.com/current/amazon/services/supported-services/cloudtrail.html#create-a-user-account)

#### Log appeared only after restarting Wazuh server

<details><summary>Evidence</summary>

![image](https://user-images.githubusercontent.com/69121070/213344455-b7ee7b6b-c61a-4887-a4e5-8118e4e2a09e.png)

</details>
